### PR TITLE
Update ReceiveProtobuf to use simpleInjector

### PR DESCRIPTION
**Pull request by Korgath**
## Description

Update ReceiveProtobuf to use simpleInjector

## References

* #0000

### CODE REVIEW

File: source/Energinet.DataHub.MarketRoles.Infrastructure/Transport/Protobuf/Integration/OneOfConfiguration.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class OneOfConfiguration<TOneOf>
 
         private Func<TOneOf, Enum>? _getMessageType;
 
-        internal OneOfConfiguration() { }
+        public OneOfConfiguration() { }
```
Comment (cmgrd) [inline source/Energinet.DataHub.MarketRoles.Infrastructure/Transport/Protobuf/Integration/OneOfConfiguration.cs:31]: Why the change to public?

File: source/Energinet.DataHub.MarketRoles.Infrastructure/Transport/Protobuf/Integration/OneOfConfiguration.cs [hunk 2]
```diff
@@ -57,7 +57,7 @@ public OneOfConfiguration<TOneOf> FromOneOf(Func<TOneOf, Enum> oneOfCase)
         /// Configure a proto buf parser
         /// </summary>
         /// <returns><see cref="ProtobufParser"/> from the configuration</returns>
-        internal ProtobufParser GetParser()
+        public ProtobufParser GetParser()
```
Comment (cmgrd) [inline source/Energinet.DataHub.MarketRoles.Infrastructure/Transport/Protobuf/Integration/OneOfConfiguration.cs:60]: Why the change to public?

File: source/Energinet.DataHub.MarketRoles.IntegrationTests/Transport/TransportTests.cs [hunk 3]
```diff
@@ -28,30 +30,33 @@ public class TransportTests
...
             // Receive
-            var receivingServiceCollection = new ServiceCollection();
-            receivingServiceCollection.ReceiveProtobuf<TestEnvelope>(
-               config => config
-                   .FromOneOf(envelope => envelope.TestMessagesCase)
-                   .WithParser(() => TestEnvelope.Parser));
-
-            var receivingServiceProvider = receivingServiceCollection.BuildServiceProvider();
-            var messageExtractor = receivingServiceProvider.GetRequiredService<MessageExtractor>();
+            var messageExtractor = container.GetInstance<MessageExtractor>();
```
Comment (cmgrd) [inline source/Energinet.DataHub.MarketRoles.IntegrationTests/Transport/TransportTests.cs:59]: This was previously organized in a sending part and a receiving part, now it's mixed. I think it was easier to follow before.

**Review by cmgrd — CHANGES_REQUESTED** [review]

**Comment by Korgath** [comment]
@cmgrd I would approve your changes, but can't since I'm the Author.. LGTM

**Review by erduna — DISMISSED** [review]

**Review by MrDach — DISMISSED** [review]

**Review by MrDach — DISMISSED** [review]

**Review by erduna — APPROVED** [review]

**Review by MrDach — APPROVED** [review]

**Review by cmgrd — APPROVED** [review]

**Pull request merged**